sql/core/src/test/scala/org/apache/spark/sql/execution/python/PythonStreamingDataSourceSuite.scala

```diff
@@ -16,15 +16,24 @@
  */
 package org.apache.spark.sql.execution.python
 
+import java.io.File
+
+import scala.concurrent.duration._
+
 import org.apache.spark.SparkException
-import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.{AnalysisException, DataFrame, Row}
 import org.apache.spark.sql.IntegratedUDFTestUtils.{createUserDefinedPythonDataSource, shouldTestPandasUDFs}
 import org.apache.spark.sql.execution.datasources.v2.python.{PythonDataSourceV2, PythonMicroBatchStream, PythonStreamingSourceOffset}
+import org.apache.spark.sql.execution.streaming.MemoryStream
 import org.apache.spark.sql.types.StructType
 import org.apache.spark.sql.util.CaseInsensitiveStringMap
 
 class PythonStreamingDataSourceSuite extends PythonDataSourceSuiteBase {
 
+  import testImplicits._
+
+  val waitTimeout = 5.seconds
```

Comment:
are 5 secs enough? I would set it to 10 or 30 secs to be sure. Streaming with Python is is pretty flaky and slow time to time.

Reply:
Changed to 15 seconds.